lib/src/client/transport/xhr_transport_stub.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2018, the gRPC project authors. Please see the AUTHORS file
+// for details. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+
+import '../options.dart';
```

Comment:
I believe the transitive dependency on dart:io will make this stub moot

Reply:
So, interestingly, dart:io imports okay on web... you just can't do anything with it. I'm curious why that is... ;)

Reply:
Yup that's weird. It could be the strict dependency platform checks don't work the same externally as I'm used to, but we definitely shouldn't rely on that behavior. Shouldn't matter if we're splitting this up, in either case.